croud/session.py

```diff
@@ -65,22 +65,21 @@ def __init__(
     ) -> JsonDict:
         url = self.url + endpoint
         resp = await self.client.post(
-            url, json={"query": query, "variables": variables}
+            url, json={"query": query, "variables": variables}, allow_redirects=False
         )
-        if resp.status == 200:
-            try:
-                result = await resp.json()
-                return result
-            except ContentTypeError:
-                print_error(
-                    "Unauthorized. Use `croud login` to login to CrateDB Cloud."
-                )
-                await self.client.close()
-                exit(1)
-        else:
-            raise Exception(
-                f"Query failed to run by returning code of {resp.status}. {query}"
-            )
+
+        if resp.status == 302:  # login redirect
+            print_error("Unauthorized. Use `croud login` to login to CrateDB Cloud.")
+            await self.client.close()
+            exit(1)
+
+        if resp.status != 200:
+            print_info(f"Query failed to run by returning code of {resp.status}.")
+            print_info(query)
+            if variables:
+                print_info(str(variables))
```

Comment:
For even nicer output we could json dump the dict:
```suggestion
                print_info(json.dumps(variables, indent=2))
```